build-distribution/action.yml

```
@@ -34,7 +34,7 @@ runs:
         python3 -m build
 
     - name: Store the distribution packages
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
         name: python-package-distributions
```

Is this what we should change to?

```suggestion
        name: python-package-distributions-${{ matrix.runs-on }}
```

---

I am fine with these changes to the names if any usages of the name are unaffected. Otherwise, consider [overwrite: true](https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md#overwriting-an-artifact) instead. It seems like this might be the closest to the old behavior, but I'm not certain what the consequences of "an entirely new artifacts" means in this context.